### PR TITLE
Some modernisation. 

### PR DESCRIPTION
**Pull request by brett-smith**
This came about due to a need for a fully JPMS compliant library, i.e. the addition of a `module-info.java`. This is supported by Java 9 and later, but to be able to make a jar compatible with earlier versions we need to use an MRJAR (multi-release). This is supported in Java 8 and onwards, so that must become the lowest common denominator.

For this reason, I have made this a major version bump. The change also necessitated updating some Maven plugins and dependencies including JUnit. As a result of updating JUnit, many deprecations needed to be fixed. The source is now warning free (with the need for a few suppression annotations).

To recap,

 * Minimum Java version increased to Java 8.
 * A Multi-release Jar is built with JPMS support. The module name is the same as the package, `com.github.jgonian.ipmath`.
 * Various Maven plugins updated.
 * Testing dependencies updated.
 * Resulting deprecations fixed.

### CODE REVIEW

**Comment by jgonian** [comment]
Hi @brett-smith, many thanks for the pull-request. On a first pass it looks good. I'll try to find some time to wrap it up and cut the new release. It'd be nice if you could also add a note in the README about versioning and Java 8.

**Comment by brett-smith** [comment]
Hi @jgonian, and thanks for the fast response. I'll get on to that now! 

On a side note, I should point out that there were no changes to `commons-ip-math-gwt`. I have no experience with GWT, but I presume nothing needs to happen with that, as it contains no classes.
